docs/source/deployorderer/ordererchecklist.md

```diff
@@ -0,0 +1,307 @@
+
+This checklist covers key configuration parameters for setting up a production ordering service. Of course, you can always refer to the orderer.yaml file for additional parameters or more information. It also provides guidance on which parameters should be overridden. The list of parameters that you need to understand and that are described in this topic include:
+
+* [General.ListenAddress](#general-listenaddress)
+* [General.ListenPort](#general-listenport)
+* [General.TLS.*](#general-tls)
+* [General.Keepalive.*](#general-keepalive)
+* [General.Cluster.*](#general-cluster)
+* [General.BoostrapMethod](#general-bootstrapmethod)
+* [General.BoostrapFile](#general-bootstrapfile)
+* [General.LocalMSPDir](#general-localmspdir)
+* [General.LocalMSPID](#general-localmspid)
+* [FileLedger.Location](#fileledger-location)
+* [Operations.*](#operations)
+* [Metrics.Provider](#metrics-provider)
+* [Metrics.Statsd.*](#metrics-statsd)
+* [Consensus.*](#consensus)
+
+## General.ListenAddress
+
+```
+# Listen address: The IP on which to bind to listen.
+ListenAddress: 127.0.0.1
+```
+
+* **`ListenAddress`**: (default value should be overridden) This is the location where the orderer will listen, for example, `0.0.0.0`. Note: unlike the peer, the `orderer.yaml` separates the address and the port, hence the [General.ListenPort](#general-listenport) parameter.
+
+## General.ListenPort
+
+```
+# Listen port: The port on which to bind to listen.
+ListenPort: 7050
+```
+
+* **`ListenPort`**: (default value should be overridden) This is the port that the orderer listens on.
+
+## General.TLS
+
+```
+Enabled: false
+# PrivateKey governs the file location of the private key of the TLS certificate.
+PrivateKey: tls/server.key
+# Certificate governs the file location of the server TLS certificate.
+Certificate: tls/server.crt
+RootCAs:
+  - tls/ca.crt
+ClientAuthRequired: false
+ClientRootCAs:
+```
+
+* **`Enabled`**: (default value should be overridden) In a production network, you should be using TLS-secured communications. This value should be `true`.
+* **`PrivateKey`**: (default value should be overridden). Provide the path to and file name of the private key generated by your TLS CA for this node.
+* **`Certificate`**: (default value should be overridden) Provide the path to and filename of the public certificate (also known as the sign certificate) generated by your TLS CA for this node.
+* **`RootCAs`**: (should be commented out) This parameter needs to be unset for normal use. It can be used to augment the orderer TLS certificate in the channel configuration orderer `MSPDir/tlscacerts` folder.
```

Comment:
As I recall, the language that was here was suggested by Jason in the google doc this doc was created from. Also I'm not really sure how this one field in orderer.yaml can augment a set of TLS certs from multiple MSPs?

Reply:
As I recall, the language that was here was suggested by Jason in the google doc this doc was created from. Also I'm not really sure how this one field in orderer.yaml can augment a set of TLS certs from multiple MSPs?

Reply:
Let's take test network as an example. There are three orgs in the channel config (orderer org, org1, org2). Each of these orgs has a tls ca cert in their msp/tlscacerts directory that gets encoded into the channel config. This set of three tls ca certs are used to verify connections.  Now, at a later time, suppose one of those certs gets expired and the associated org is not able to immediately update their tls ca cert in the channel config. That org could provide a new tls ca cert, and it could be included in orderer.yaml for this orderer to augment the list of three from the channel config. So now it has a list of four (3 from channel config, one from orderer.yaml). This is consistent with what Jason said in the google doc - his main point was that any cert here would augment instead of override the list from the channel config.

This is a rare scenario, which is why we say it should be commented out, even in the sample orderer.yaml.

(Furthermore, I don't think this list is even used anymore, but that's a different story).

Reply:
+1 I think I may have used a '/' notation meaning "or", I did not mean an explicit path.  Dave's wording is correct.